src/main/java/org/teacon/slides/ProjectorRenderer.java

```diff
@@ -43,10 +60,63 @@ public void render(ProjectorTileEntity tile, float partialTicks, MatrixStack mat
         matrixStack.pop();*/
 
         matrixStack.pop();
+
+        if (ClientTick.renderOutline) {
+            matrixStack.push();
+
+            Minecraft mc = Minecraft.getInstance();
+            BlockRendererDispatcher renderer = mc.getBlockRendererDispatcher();
+            OutlineLayerBuffer outline = mc.getRenderTypeBuffers().getOutlineBufferSource();
+
+            outline.setColor(255, 255, 255, 255);
+            renderer.renderBlock(tile.getBlockState(), matrixStack, outline, combinedLight, combinedOverlay, EmptyModelData.INSTANCE);
+
+            matrixStack.pop();
+        }
     }
 
     @Override
     public boolean isGlobalRenderer(ProjectorTileEntity tile) {
         return true;
     }
+
+    @Mod.EventBusSubscriber(modid = "slide_show", value = Dist.CLIENT, bus = Mod.EventBusSubscriber.Bus.FORGE)
+    public static final class ClientTick {
+
+        private static boolean renderOutline = false;
+
+        private static boolean isProjector(ItemStack i) {
+            return SlideShow.projector.asItem().equals(i.getItem());
+        }
+
+        @SubscribeEvent
+        public static void tick(TickEvent.ClientTickEvent event) {
+            if (event.phase == TickEvent.Phase.END) {
+                Minecraft mc = Minecraft.getInstance();
+                if (mc.world != null && mc.playerController.getCurrentGameType() == GameType.CREATIVE) {
+                    renderOutline = Streams.stream(mc.player.getHeldEquipment()).anyMatch(ClientTick::isProjector);
+                }
+            }
+        }
+
+        private static Entity createDummyEntity(ClientWorld world) {
+            // dirty hack - create a long-lived glowing entity to activate entity outline shader
+            Entity entity = new AreaEffectCloudEntity(world, 0.0, 0.0, 0.0);
+            entity.setBoundingBox(IForgeTileEntity.INFINITE_EXTENT_AABB);
+            entity.ignoreFrustumCheck = true;
+            entity.setInvisible(true);
+            entity.setGlowing(true);
+            entity.canUpdate(false);
+            return entity;
```

Comment:
Also make it invulnerable by calling `setInvulnerable(true)`? Will `/kill @e` affect this?

Reply:
`/kill @e` is a server command and affects server entities only.

However, it should indeed be attentioned whether it is possible that the client entity will come to death.